pkg/user/password_reset.go

```diff
@@ -0,0 +1,27 @@
+package user
+
+import (
+	"time"
+
+	"github.com/dgryski/trifles/uuid"
+)
+
+type PasswordReset struct {
+	Token     string
+	CreatedAt time.Time
+}
+
+func NewPasswordReset() *PasswordReset {
+	return &PasswordReset{
+		Token:     generateToken(),
+		CreatedAt: time.Now(),
+	}
+}
```

Comment:
You can add another func for conversion between domain model and mongo doc:

```go
func PasswordResetFrom(token string, createdAt time.Time) *PasswordReset {
	return &PasswordReset{
		Token:     token,
		CreatedAt: createdAt,
	}
}
```

Of course, unit test is required